### PR TITLE
Participant scenario testing seeds

### PR DESCRIPTION
**Pull request by mattonfoot**
### Context

Create see scenarios for specific participant record states and check that they can be presented to and found by the appropriate users using our UIs and APIs. 

### Changes proposed in this pull request

- Add reusable participant seeds for the various scenarios we know of
- Add some reusable shared_examples to test the scenarios in various UIs and APIs

#### Bug fixes

- Fixes where the header on the admin participant details pages get the cohort start year

### CODE REVIEW

File: app/helpers/admin_helper.rb [hunk 1]
```diff
@@ -49,11 +49,15 @@ def format_address(*parts)
...
 
-  def admin_participant_header_and_title(profile:, section:)
+  def admin_participant_header_and_title(presenter:, section:)
+    profile = presenter.participant_profile
     full_name = profile.full_name
     role = admin_participant_role_name(profile.class.name)
     trn = profile.teacher_profile.trn
-    cohort = profile.cohort&.start_year
+
+    # TODO: There should be a single consistent way to find the relevant cohort location
```
Comment (peteryates) [inline app/helpers/admin_helper.rb:58]: There really should.

File: db/new_seeds/scenarios/school_cohorts/fip.rb [hunk 2]
```diff
@@ -29,12 +29,43 @@ def with_programme(**args)
...
+          @partnership = FactoryBot.create(:seed_partnership, *traits, **options)
+
+          FactoryBot.create(
+            :seed_provider_relationship,
+            cohort:,
+            lead_provider: partnership.lead_provider,
+            delivery_partner: partnership.delivery_partner,
+          )
+
+          @partnership
```
Comment (peteryates) [inline db/new_seeds/scenarios/school_cohorts/fip.rb:66]: Should this be returning `self` so it's chainable?

**Review by peteryates — APPROVED** [review]
Looks good, all seems sensible. Left a minor commeont/suggestion.

File: db/new_seeds/scenarios/school_cohorts/fip.rb (hunk 2 above)
Comment (mattonfoot) [reply db/new_seeds/scenarios/school_cohorts/fip.rb:66]: from what I saw in others the pattern was that "with_XXX" is chainable and "add_XXX" returns the thing that has been added?

File: app/helpers/admin_helper.rb (hunk 1 above)
Comment (mattonfoot) [reply app/helpers/admin_helper.rb:58]: moving it to the presenter at least is a start I guess

**Pull request merged**